.github/workflows/formatting.yml

```diff
@@ -19,6 +19,6 @@ jobs:
         - name: install dependencies
           run: |
             sudo apt-get update -y -qq
-            sudo apt-get install -y --force-yes -qq git clang-format-12
+            sudo apt-get install -y --force-yes -qq git clang-format-17
```

Comment:
Any suggestions for how to do this if `apt-get` doesn't find it?

Reply:
Try updating to `ubuntu-24.04`

Reply:
You might need to add the LLVM repos explicitly. See https://apt.llvm.org/